src/plugins/legacy/reformat/medCropToolBox.cpp

```diff
@@ -445,36 +459,40 @@ void medCropToolBoxPrivate::applyInverseOrientationMatrix(double *worldPointIn,
     std::copy(homogeneousVector, homogeneousVector + 3, worldPointOut);
 }
 
-void medCropToolBoxPrivate::generateOutput()
+bool medCropToolBox::generateOutput()
 {
+    bool res = medAbstractProcessLegacy::SUCCESS;
+
     double minBoxCorner[3], maxBoxCorner[3];
     int minIndices[3], maxIndices[3];
 
-    getMinAndMaxBoxCorners(minBoxCorner, maxBoxCorner);
-    applyOrientationMatrix(minBoxCorner, minBoxCorner);
-    applyOrientationMatrix(maxBoxCorner, maxBoxCorner);
-    view3D->GetImageCoordinatesFromWorldCoordinates(minBoxCorner, minIndices);
-    view3D->GetImageCoordinatesFromWorldCoordinates(maxBoxCorner, maxIndices);
+    d->getMinAndMaxBoxCorners(minBoxCorner, maxBoxCorner);
+    d->applyOrientationMatrix(minBoxCorner, minBoxCorner);
+    d->applyOrientationMatrix(maxBoxCorner, maxBoxCorner);
+    d->view3D->GetImageCoordinatesFromWorldCoordinates(minBoxCorner, minIndices);
+    d->view3D->GetImageCoordinatesFromWorldCoordinates(maxBoxCorner, maxIndices);
 
-    outputData.clear();
+    d->outputData.clear();
 
-    for (unsigned int layer = 0; layer < view->layersCount(); layer++)
+    for (unsigned int layer = 0; layer < d->view->layersCount(); layer++)
     {
-        medAbstractData *imageData = view->layerData(layer);
+        medAbstractData *imageData = d->view->layerData(layer);
         medAbstractData *output = nullptr;
 
         if (DISPATCH_ON_3D_PIXEL_TYPE(&medCropToolBoxPrivate::extractCroppedImage,
-                                      this, imageData, minIndices, maxIndices, &output)
+                                      d, imageData, minIndices, maxIndices, &output)
                 == medAbstractProcessLegacy::SUCCESS)
         {
-            outputData.append(output);
+            d->outputData.append(output);
         }
         else
         {
-            medMessageController::instance()->showError("Drop a 3D volume in the view", 3000);
-            qDebug()<<__FILE__<<":"<<__LINE__<<imageData->identifier();
+            handleDisplayError(medAbstractProcessLegacy::DIMENSION_3D);
+            res = medAbstractProcessLegacy::FAILURE;
```

Comment:
It would be better to use the error code actually returned by `DISPATCH_ON_3D_PIXEL_TYPE` in the call to `handleDisplayError` and also when returning the error from this function.

Reply:
indeed, i updated the PR with a better error code management